### PR TITLE
Feedback Changes 10/12

### PR DESCRIPTION
**Pull request by openbots-ff**

### CODE REVIEW

File: OpenBots.Commands/OpenBots.Commands.Outlook/GetOutlookEmailsCommand.cs [hunk 1]
```diff
@@ -25,7 +25,7 @@ public class GetOutlookEmailsCommand : ScriptCommand
         [PropertyDescription("Source Mail Folder Name")]
         [InputSpecification("Enter the name of the Outlook mail folder the emails are located in.")]
         [SampleUsage("Inbox || {vFolderName}")]
-        [Remarks("")]
+        [Remarks("Source folder cannot be a sub mail folder.")]
```
Comment (openbots-ch) [inline OpenBots.Commands/OpenBots.Commands.Outlook/GetOutlookEmailsCommand.cs:28]: Lets reword this to "cannot be a subfolder." if that's the intent.

**Review by openbots-ch — CHANGES_REQUESTED** [review]

File: OpenBots.Commands/OpenBots.Commands.DataTable/FilterDataTableCommand.cs [hunk 2]
```diff
@@ -63,40 +61,40 @@ public override void RunCommand(object sender)
             foreach (string item in listPairs)
             {
                 string temp;
-                temp = item.Trim('(');
+                temp = item.Trim().TrimStart(',').TrimStart('(');
                 var tempList = temp.Split(',');
                 t.Insert(i, Tuple.Create(tempList[0], tempList[1]));
                 i++;
             }
```
[the remark this answers is not in the document]
Comment (openbots-ff) [reply OpenBots.Commands/OpenBots.Commands.DataTable/FilterDataTableCommand.cs:68]: I don't this this is reused anywhere at the moment. I'll put it in core if it ever is, but I might replace this method altogether soon when I redesign the way filtering works (using a datagridview)

File: OpenBots.Commands/OpenBots.Commands.Outlook/GetOutlookEmailsCommand.cs (hunk 1 above)
Comment (openbots-ff) [reply OpenBots.Commands/OpenBots.Commands.Outlook/GetOutlookEmailsCommand.cs]: done

File: OpenBots.Commands/OpenBots.Commands.Outlook/MoveCopyOutlookEmailCommand.cs [hunk 3]
```diff
@@ -30,7 +30,7 @@ public class MoveCopyOutlookEmailCommand : ScriptCommand
         [PropertyDescription("Destination Mail Folder Name")]
         [InputSpecification("Enter the name of the Outlook mail folder the emails are being moved/copied to.")]
         [SampleUsage("New Folder || {vFolderName}")]
-        [Remarks("")]
+        [Remarks("Destination folder cannot be a sub mail folder.")]
```
[the remark this answers is not in the document]
Comment (openbots-ff) [reply OpenBots.Commands/OpenBots.Commands.Outlook/MoveCopyOutlookEmailCommand.cs]: done

**Review by openbots-ch — DISMISSED** [review]

File: OpenBots.Commands/OpenBots.Commands.Excel/ExcelAppendSheetCommand.cs [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+using System;
+using System.Collections.Generic;
+using System.Windows.Forms;
+using Application = Microsoft.Office.Interop.Excel.Application;
+
+namespace OpenBots.Commands.Excel
+{
+    [Serializable]
+    [Group("Excel Commands")]
+    [Description("This command activates a specific Worksheet in an Excel Workbook.")]
```
Comment (openbots-ab) [inline OpenBots.Commands/OpenBots.Commands.Excel/ExcelAppendSheetCommand.cs:18]: Please update the description.

File: OpenBots.Commands/OpenBots.Commands.Excel/ExcelDeleteRangeCommand.cs [hunk 5]
```diff
@@ -34,7 +34,7 @@ public class ExcelDeleteRangeCommand : ScriptCommand
         [PropertyDescription("Shift Cells Up")]
         [PropertyUISelectionOption("Yes")]
         [PropertyUISelectionOption("No")]
-        [InputSpecification("Indicate whether the row(s) below will be shifted up to replace the old row(s).")]
+        [InputSpecification("'Yes' removes the entire row. 'No' only clears the row of its cell values.")]
```
Comment (openbots-ab) [inline OpenBots.Commands/OpenBots.Commands.Excel/ExcelDeleteRangeCommand.cs:37]:
```suggestion
        [InputSpecification("'Yes' removes the entire range. 'No' only clears the range of its cell values.")]
```

File: OpenBots.Commands/OpenBots.Commands.List/RemoveListItemCommand.cs [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+using System.Collections.Generic;
+using System.Data;
+using System.Windows.Forms;
+using Exception = System.Exception;
+
+namespace OpenBots.Commands.List
+{
+    [Serializable]
+    [Group("List Commands")]
+    [Description("This command updates an item in an existing List variable at a specified index.")]
```
Comment (openbots-ab) [inline OpenBots.Commands/OpenBots.Commands.List/RemoveListItemCommand.cs:21]: Please update the description

File: OpenBots.Engine/AutomationEngineInstance.cs [hunk 7]
```diff
@@ -346,12 +343,12 @@ public void ExecuteCommand(ScriptAction command)
                     (parentCommand.CommandName == "BeginRetryCommand" || (parentCommand.CommandName == "BeginSwitchCommand")))
                 {
                     //run the command and pass bgw/command as this command will recursively call this method for sub commands
-                    command.IsExceptionIgnored = true;
+                    //command.IsExceptionIgnored = true;
```
Comment (openbots-ab) [inline OpenBots.Engine/AutomationEngineInstance.cs:346]: Please remove this comment.

File: OpenBots.Engine/AutomationEngineInstance.cs [hunk 8]
```diff
@@ -346,12 +343,12 @@ public void ExecuteCommand(ScriptAction command)
...
                     parentCommand.RunCommand(this, command);
                 }
                 else if (parentCommand.CommandName == "SequenceCommand")
                 {
-                    command.IsExceptionIgnored = true;
+                    //command.IsExceptionIgnored = true;
```
Comment (openbots-ab) [inline OpenBots.Engine/AutomationEngineInstance.cs:351]: Please remove this comment.

**Review by openbots-ab — CHANGES_REQUESTED** [review]
Please make a few requested changes

File: OpenBots.Commands/OpenBots.Commands.Excel/ExcelAppendSheetCommand.cs (hunk 4 above)
Comment (openbots-ff) [reply OpenBots.Commands/OpenBots.Commands.Excel/ExcelAppendSheetCommand.cs:18]: done

File: OpenBots.Commands/OpenBots.Commands.Excel/ExcelDeleteRangeCommand.cs (hunk 5 above)
Comment (openbots-ff) [reply OpenBots.Commands/OpenBots.Commands.Excel/ExcelDeleteRangeCommand.cs:37]: done

File: OpenBots.Commands/OpenBots.Commands.Excel/ExcelRenameSheetCommand.cs [hunk 9]
```diff
@@ -0,0 +1,79 @@
...
+using System;
+using System.Collections.Generic;
+using System.Windows.Forms;
+using Application = Microsoft.Office.Interop.Excel.Application;
+
+namespace OpenBots.Commands.Excel
+{
+    [Serializable]
+    [Group("Excel Commands")]
+    [Description("This command activates a specific Worksheet in an Excel Workbook.")]
```
[the remark this answers is not in the document]
Comment (openbots-ff) [reply OpenBots.Commands/OpenBots.Commands.Excel/ExcelRenameSheetCommand.cs:18]: done

File: OpenBots.Commands/OpenBots.Commands.List/RemoveListItemCommand.cs (hunk 6 above)
Comment (openbots-ff) [reply OpenBots.Commands/OpenBots.Commands.List/RemoveListItemCommand.cs:21]: done

File: OpenBots.Engine/AutomationEngineInstance.cs (hunk 7 above)
Comment (openbots-ff) [reply OpenBots.Engine/AutomationEngineInstance.cs:346]: I'm keeping it for now because it solved a bug but I can't remember why it was needed in the first place.

File: OpenBots.Engine/AutomationEngineInstance.cs (hunk 8 above)
Comment (openbots-ff) [reply OpenBots.Engine/AutomationEngineInstance.cs:351]: same as above

**Review by openbots-ch — DISMISSED** [review]

**Review by openbots-ab — APPROVED** [review]

**Review by openbots-ch — APPROVED** [review]
